packages/core/test/useBlockNumber.test.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import { expect } from 'chai'
+import { renderHook } from '@testing-library/react-hooks'
+import { useBlockNumber, BlockNumberProvider } from '../src'
+import { TestConnector, Web3Wrapper, waitUntil } from './utils'
+import { MockProvider } from '@ethereum-waffle/provider'
+
+describe('useBlockNumber', () => {
+  it('retrieves block number', async () => {
+    const {result, waitForValueToChange} = renderHook(useBlockNumber, {
+      wrapper: ({children}) => <Web3Wrapper><BlockNumberProvider>{children}</BlockNumberProvider></Web3Wrapper>
+    })
```

Comment:
Idea to discuss:
```ts
    const {result, waitForValueToChange} = renderWeb3Hook(useBlockNumber)
```

`renderWeb3Hook` already wrapped it into standard Providers (web3, block, etc)

Yay oe ney?

Reply:
```tsx
describe('useBlock', () => {
  it('fresh', () => {
    const {result, context} = renderWeb3Hook(useBlockNumber);
    expect(result).to.eq(0)
  });

  it('mine block', () => {
    const {context} = renderWeb3Hook(useBlockNumber);
    mineBlock(context);
    expect(result).to.afterRender.eq(1)
  });

}
```